models/buffer_disk.go

```diff
@@ -293,5 +292,6 @@ func (b *DiskBuffer) handleEmptyFile() {
 		log.Printf("E! readIndex: %d, buffer len: %d", b.readIndex(), b.length())
 		panic(err)
 	}
+	b.mask = b.mask[1:]
```

Comment:
@srebhan Adding my understanding of the issue here to clarify the importance of this line:

The line `b.mask = b.mask[1:]` correctly adjusts the mask after truncation. Here's what happens:

* **Before truncation**:

  * The WAL file contains a dummy metric at index 0.
  * The mask is `[0]`, indicating the dummy metric should be skipped.

* **After truncation and this fix**:

  * The dummy metric is removed from the WAL.
  * The mask is correctly reset to `[]`.
  * The new metric at index 0 is **not masked** and will be processed as expected.

This change seems critical to ensure new data isn’t mistakenly skipped. Let me know if I’ve misunderstood anything.